scripts/scil_fix_dsi_studio_trk.py

```diff
@@ -0,0 +1,205 @@
+                                 MutualInformationMetric,
+                                 AffineRegistration)
+from dipy.align.transforms import RigidTransform3D
+from dipy.io.stateful_tractogram import StatefulTractogram, Space
+from dipy.io.utils import get_reference_info
+from dipy.io.streamline import save_tractogram, load_tractogram
+from dipy.reconst.utils import _roi_in_volume, _mask_from_roi
+import nibabel as nib
+import numpy as np
+
+from scilpy.io.utils import (add_overwrite_arg,
+                             assert_inputs_exist,
+                             assert_outputs_exist)
+from scilpy.utils.streamlines import (transform_warp_streamlines,
+                                      cut_invalid_streamlines)
+from scilpy.utils.transformation import flip_sft
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawTextHelpFormatter)
+
+    p.add_argument('in_dsi_tractogram',
+                   help='Path of the input tractogram file from DSI studio '
+                        '(.trk).')
+    p.add_argument('in_dsi_fa',
+                   help='Path of the input FA from DSI Studio (.nii.gz).')
+    p.add_argument('out_tractogram',
+                   help='Path of the output tractogram file.')
+    p.add_argument('--in_native_fa',
+                   help='Path of the input FA from Dipy/MRtrix (.nii.gz).\n'
+                        'Move the tractogram back to a "proper" space, include'
+                        'registration.')
+    p.add_argument('--auto_crop', action='store_true',
+                   help='If both FA are not already BET, perform registration \n'
+                        'using a centered-cube crop to ignore the skull.\n'
+                        'A good BET for both is more robust.')
+    transfo = p.add_mutually_exclusive_group()
+    transfo.add_argument('--save_transfo', metavar='FILE',
+                         help='Save estimated transformation to avoid '
+                              'recomputing (.txt).')
+    transfo.add_argument('--load_transfo', metavar='FILE',
+                         help='Load estimated transformation to apply to other '
+                              'files (.txt).')
+    invalid = p.add_mutually_exclusive_group()
+    invalid.add_argument('--cut_invalid', action='store_true',
+                         help='Cut invalid streamlines rather than removing '
+                              'them.\nKeep the longest segment only.')
+    invalid.add_argument('--remove_invalid', action='store_true',
+                         help='Remove the streamlines landing out of the '
+                              'bounding box.')
+    invalid.add_argument('--keep_invalid', action='store_true',
+                         help='Keep the streamlines landing out of the '
+                              'bounding box.')
+    add_overwrite_arg(p)
+
+    return p
+
+
+def get_axis_shift_vector(flip_axes):
```

Comment:
why do you have this here instead of transformation.py

Reply:
get_axis_flip_vector

Reply:
It is a piece of code unique to DSI data correction and there is no use for it outside (for now). Since this is risky spatial transform I would leave it there.

Also it use the flip_axes variable, but it is used as a shift vector (in a -= statement below)